azure/services/virtualmachines/virtualmachines.go

```diff
@@ -128,7 +128,10 @@ func (s *Service) getAddresses(ctx context.Context, vm compute.VirtualMachine, r
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "virtualmachines.Service.getAddresses")
 	defer done()
 
-	var addresses []corev1.NodeAddress
+	addresses := []corev1.NodeAddress{{
+		Type:    corev1.NodeInternalDNS,
+		Address: *vm.Name,
```

Comment:
@jsturtevant is this also accurate for Windows VMs?

Reply:
I would think that the DNS setup inside the vnet is the same regardless of the OS of the instance?

Reply:
the dns works the same but we do some funny things with the [windows name](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b97fbf54f0b919e42f3c21eba7f0f12897623883/azure/scope/machine.go#L349-L352).  

This scope name is the name that gets assigned to the Azure VM: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b97fbf54f0b919e42f3c21eba7f0f12897623883/azure/services/virtualmachines/virtualmachines.go#L78-L80  

so we are good.